app/src/org/commcare/services/CommCareSessionService.java

```diff
@@ -236,8 +246,34 @@ public void showLoggedInNotification(@Nullable User user) {
             notificationBuilder.setContentText(contentText);
         }
 
-        //Send the notification.
-        this.startForeground(NOTIFICATION, notificationBuilder.build());
+        //Send the notification
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && CommCareApplication.isAppInBackground())

```

Comment:
Have you been able to repro the crash with the foreground notification here ? I would think that since the foreground service has already started the app will always be in the foreground.